### PR TITLE
Implement zooming and panning in the profiler

### PR DESCRIPTION
**Pull request by reach-satori**
As mentioned in godotengine/godot-proposals#2045 and the quite old #5682. Uses mouse wheel for zooming and middle click drag for panning. Little demonstration included below.

https://user-images.githubusercontent.com/30383615/232031698-691542cb-de74-4dbc-ba39-547ebf2beb28.mp4

### CODE REVIEW

**Comment by avedar** [comment]
I think being able to click and drag to highlight/select a particular area of the graph and have it zoom into that section would be an excellent addition to these changes. 
Thanks for the improvements regardless!

**Comment by jamesmcm** [comment]
This would be really helpful, is there a reason it wasn't merged for 4.x so far?

**Comment by AThousandShips** [comment]
The improvement hasn't been decided on and this hasn't been reviewed yet, might be once 4.2 is released and new features are focused on once again

**Review by Calinou — COMMENTED** [review]
Tested locally (rebased on top of `master` 6afd320984cf14198368cc6c53752813a02169e3), it works as expected. Code looks good to me at a glance.

I suggest making panning work with the right mouse button in addition to the middle mouse button, so that it works on laptop trackpads out of the box too (where middle-clicking is usually not possible easily).

PS: We should really add a grid to the profiler (and visual profiler) backgrounds, so that the user is more aware of the current zoom level :slightly_smiling_face:

File: editor/debugger/editor_profiler.cpp [hunk 1]
```diff
@@ -502,9 +509,34 @@ void EditorProfiler::_graph_tex_input(const Ref<InputEvent> &p_ev) {
 				frame_delay->start();
 			}
 		}
+	}
+
+	if (mb.is_valid() && (mm->get_button_mask().has_flag(MouseButtonMask::MIDDLE) || mm->get_button_mask().has_flag(MouseButtonMask::RIGHT)) && graph_zoom > 0) {
```
Comment (AThousandShips) [inline editor/debugger/editor_profiler.cpp:514]:
```suggestion
	if (graph_zoom > 0 && mb.is_valid() && (mm->get_button_mask().has_flag(MouseButtonMask::MIDDLE) || mm->get_button_mask().has_flag(MouseButtonMask::RIGHT))) {
```
Cheaper condition first

**Comment by jakkarn** [comment]
Any progress on this? It would be amazing to have! The ui is almost useless because of the microscopic frames and tiny framestep buttons.

**Comment by Calinou** [comment]
> Any progress on this? It would be amazing to have! The ui is almost useless because of the microscopic frames and tiny framestep buttons.

There are merge conflicts that need to be resolved first.

@reach-satori Could you look into rebasing this PR against the latest `master` branch? See [Pull request workflow](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions :slightly_smiling_face:

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected.

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**